### PR TITLE
Unit tests for 2.0 branches

### PR DESCRIPTION
**Pull request by morparia-p**
Fixed tests for Interface.pm and Interface.cgi

### CODE REVIEW

**Comment by julius-globalnoc** [comment]
Can one of the admins verify this patch?

File: perl-lib/OESS/lib/OESS/Interface.pm [hunk 1]
```diff
@@ -86,24 +86,24 @@ sub to_hash{
...
 sub _fetch_from_db{
     my $self = shift;
-
+    $self->{'logger'}->error("Inside fetch from db");
     if (!defined $self->{'interface_id'}) {
         if (defined $self->{'name'} && defined $self->{'node'}) {
             my $interface_id = OESS::DB::Interface::get_interface(db => $self->{'db'}, interface => $self->{'name'}, node => $self->{'node'});
             if (!defined $interface_id) {
-                $self->{'logger'}->error("Unable to fetch interface $self->{name} on $self->{node} from the db!");
+                $self->{'logger'}->error("sdfsdfUnable to fetch interface $self->{name} on $self->{node} from the db!");
```
Comment (ajragusa) [inline perl-lib/OESS/lib/OESS/Interface.pm]: what is this?

Comment (ajragusa) [reply perl-lib/OESS/lib/OESS/Interface.pm]: I see several places where the logs have had some text appended, could you clean these up.

File: perl-lib/OESS/lib/OESS/Interface_test.t [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+use OESS::DB::Interface;
+
+$OESS::Database::ENABLE_DEVEL=1;
+Log::Log4perl->init("/etc/oess/logging.conf");
+my $db = OESS::Database->new();
+my $interface = OESS::Interface->new(	db 	=> $db,
+					name	=> "asd",
+					node	=> 
+						);
+ok(defined($interface),"Object interface initiated");
```
Comment (ajragusa) [inline perl-lib/OESS/lib/OESS/Interface_test.t]: There should be way more tests we can do on the interface object itself.  For example fetching the name, checking to see if it has a cloud_interconnect_id, getting a list of ACL, checking to see if the VLAN is in use.

**Comment by ajragusa** [comment]
test to see if still goes to julius

File: perl-lib/OESS/lib/OESS/Interface.pm (hunk 1 above)
Comment (morparia-p) [reply perl-lib/OESS/lib/OESS/Interface.pm]: Sure!

File: perl-lib/OESS/lib/OESS/Interface_test.t (hunk 2 above)
Comment (morparia-p) [reply perl-lib/OESS/lib/OESS/Interface_test.t]: On it!

**Comment by ajragusa** [comment]
@morparia-p I tried to run these tests on my local box and ran into a few issues... the first of which was I could not load the known state database properly.... 

You have to lock the tables before you can do an insert... so in the known state before you do updates to the tables you need to lock them

LOCK TABLES `circuit_instantiation` WRITE , `interface` WRITE, `interface_acl` WRITE , `entity` WRITE , `vrf` WRITE , `vrf_ep` WRITE , `cloud_connection_vrf_ep` WRITE;

that is what I added to your known state to make it work

File: perl-lib/OESS/t/Interface_test.t [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+use OESS::Interface;
+use Test::More tests => 30;
+use Log::Log4perl;
+use OESS::DB;
+use Data::Dumper;
+use Test::Deep;
+
+
+# Initialize logging
+Log::Log4perl->init("/etc/oess/logging.conf");
```
Comment (ajragusa) [inline perl-lib/OESS/t/Interface_test.t]: You can use this for testing but not for actual tests (this file does not exist when running the test suite)

File: perl-lib/OESS/t/Interface_test.t [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+# Initialize DB
+my $db = OESS::DB->new();
```
Comment (ajragusa) [inline perl-lib/OESS/t/Interface_test.t]: we need to make this use the test config file not the one in /etc/oess/database.xml

adding the following to the use statements will help

use FindBin;
my $path;

BEGIN {
    if($FindBin::Bin =~ /(.*)/){
            $path = $1;
      }
}

use lib "$path";
use OESSDatabaseTester;

then you can do 

my $db = OESS::DB->new(config => OESSDatabaseTester::getConfigFilePath());

basically replace /etc/oess/database.xml with `OESSDatabaseTester::getConfigFilePath()` everywhere

File: perl-lib/OESS/t/Interface_test.t [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+ok($interface->cloud_interconnect_id() eq 'Test',"The method cloud_interconnect_id() gives expected output for interface_id 391");
+$a = $interface->cloud_interconnect_type();
+ok($a eq undef , "The method cloud_interconnect_type() returns correct information");
+
+ok($interface->description() eq 'e15/1', "The method description() returns correct information");
+
+#ok($interface->port_number() eq $test_interface->{'port_number'}, "The method port_number() returns correct information");
+
+cmp_deeply($interface->acls()->{'acls'} ,
```
Comment (ajragusa) [inline perl-lib/OESS/t/Interface_test.t]: this does not seem to line up at all....

**Comment by ajragusa** [comment]
lets ignore the CGI ones for now... I will work on getting the apache test stuff included later on... we have enough to fix in the library test right now...

**Comment by jonstout** [comment]
Closing due to inactivity

**Pull request closed without merging**